File: trail2/그 계절, 그 날/that-season-that-day.cpp
#include <iostream>

using namespace std;

int Y, M, D;

string getSeason() {
    if (3 <= M && M <= 5)
        return "Spring";
    else if (6 <= M && M <= 8)
        return "Summer";
    else if (9 <= M && M <= 11)
        return "Fall";
    
    return "Winter";
}

bool isLeanYear() {
    if (Y % 4 == 0) {
        if (Y % 100 != 0)
            return true;

        if (Y % 100 == 0 && Y % 400 == 0)
            return true;
    }

    return false;
}

int getLastDay() {
    if (M == 2) {
        if (isLeanYear())
            return 29;
        else
            return 28;
    } else if (M == 4 || M == 6 || M == 9 || M == 11) {
        return 30;
    }

    return 31;
}
 
bool judgeDay() {
    if (M <= 12 && D <= getLastDay())
        return true;
    
    return false;
}

int main() {
    cin >> Y >> M >> D;

    // Please write your code here.
    if (judgeDay())
        cout << getSeason() << endl;
    else
        cout << -1 << endl;

    return 0;
}

File: trail2/그 계절, 그 날/that-season-that-day.py
y, m, d = tuple(map(int, input().split()))

def is_yoon_year():
    if (y % 4 != 0 or y % 100 == 0 and y % 400 != 0):
        return False
    return True

def is_date():
    if (m == 2):
        if (is_yoon_year()):
            if (d >= 1 and d <= 29):
                return True
        else:
            if (d >= 1 and d <= 28):
                return True
    elif (m == 1 or m == 3 or m == 5 or m == 7 or m == 8 or m == 10 or m == 12):
        if (d >= 1 and d <= 31):
            return True
    elif (m == 4 or m == 6 or m == 9 or m == 11):
        if (d >= 1 and d <= 30):
            return True
    
    return False
    

def get_season():
    if (is_date() == False):
        return "-1"
    
    if (m >= 3 and m <= 5):
        return "Spring"
    elif (m >= 6 and m <= 8):
        return "Summer"
    elif (m >= 9 and m <= 11):
        return "Fall"
    else:
        return "Winter"

print(get_season())